### PR TITLE
create2 option

### PR DESCRIPTION
**Pull request by z0r0z**
add create2 optionality w/ salt

### CODE REVIEW

**Comment by WyseNynja** [comment]
I don't think the is_zero check on salt is the right way to do this. I currently use create2 with empty salts for almost all of my deploys. I do this because I care about easily recreating the same address and not because I want a vanity address. I could change all my code to use a salt of "1" or something as default, but I think two separate functions would be better

**Comment by strollinghome** [comment]
@wighawag @ZeframLou any chance of getting this merged? Interested in using this and deterministic address would-be killer. 

Also, I've been talking to people at Etherscan and they added minor support for detecting these types of clones already. Ideally, we'd get something similar to how EIP-1167 gets code verified automatically for all clones.

File: src/ClonesWithImmutableArgs.sol [hunk 1]
```diff
@@ -137,7 +140,12 @@ library ClonesWithImmutableArgs {
             }
             // solhint-disable-next-line no-inline-assembly
             assembly {
-                instance := create(0, ptr, creationSize)
+                if iszero(salt) {
```
Comment (wighawag) [inline src/ClonesWithImmutableArgs.sol:143]: maybe instead of this check, we should create 2 functions `clone` as before  and `cloneDeterministic`

Comment (strollinghome) [reply src/ClonesWithImmutableArgs.sol:143]: Yeah, happy to make a new PR to work on this, unless @z0r0z is up for continuing on this?

Comment (strollinghome) [reply src/ClonesWithImmutableArgs.sol:143]: Also super noob question, I'm not very good at assembly. But would there be a way to write a `predictDeterministic` function that predicts the address it'll deploy to without actually deploying the contract? I know OZ's `Clones` have that too.

**Pull request closed without merging**